app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -80,22 +91,10 @@ def fetch_signing_key
           )
         end
 
-        def authentication_parameters_class
-          @authentication_parameters_class ||= Authentication::AuthnJwt::AuthenticationParameters
-        end
-
         def extract_token_from_credentials
           @extract_token_from_credentials ||= Authentication::AuthnJwt::InputValidation::ExtractTokenFromCredentials.new
         end
 
-        def validate_and_decode_token_class
-          @validate_and_decode_token_class ||= Authentication::AuthnJwt::ValidateAndDecode::ValidateAndDecodeToken
-        end
-
-        def validate_resource_restrictions_class
-          @validate_resource_restrictions_class ||= Authentication::ResourceRestrictions::ValidateResourceRestrictions
-        end
-
         def create_identity_provider
```

Comment:
this should be moved to the method signature. also - i don't see why we need to have these log messages

Reply:
please move the following to the method signature:
- create_signing_key_interface
- fetch_jwt_claims_to_validate
- restrictions_from_annotations

maybe also `validate_resource_restrictions`

Reply:
Moved to dependency. 
@nessiLahav take it about the logs